gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -193,28 +193,42 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) error {
 		gno.MachineOptions{
 			PkgPath:   "",
 			Output:    os.Stdout, // XXX
-			Store:     store,
-			Alloc:     store.GetAllocator(),
+			Store:     gnostore,
+			Alloc:     gnostore.GetAllocator(),
 			Context:   msgCtx,
 			MaxCycles: vm.maxCycles,
 		})
-	defer m2.Release()
-	m2.RunMemPackage(memPkg, true)
+	defer func() {
```

Comment:
This defer function is duplicated in a few different places. What are you thoughts on defining a new function like `handleConsumeGasPanic` that accepts the necessary arguments?

Reply:
I tried the suggested solution. The defer handler won't return an error, which is required in the return parameters defined in the parent function. Originally, the value of err was determined through the closure in defer. To write the handler, in addition to other parameters, we must use *error as a parameter passed to the handler and then modify it in the handler function. It works, but this extra layer of abstraction makes the panic and error handling flow even harder to understand. Therefore, I decided to keep it as it is. It's a trade-off between readability and the reusability of the abstraction.